subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/ChangesByGradleFileWatchingIntegrationTest.groovy

```diff
@@ -0,0 +1,260 @@
+        then:
+        executedAndNotSkipped(":incremental")
+        file("build/output1").assertExists()
+    }
+
+    def "detects changes to manifest"() {
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+
+            jar {
+                manifest {
+                    attributes('Created-By': providers.systemProperty("creator"))
+                }
+            }
+        """
+
+        when:
+        withWatchFs().run "jar", "-Dcreator=first"
+        then:
+        file("build/tmp/jar/MANIFEST.MF").text.contains("first")
+        executedAndNotSkipped(":jar")
+
+        when:
+        withWatchFs().run "jar", "-Dcreator=second"
+        then:
+        file("build/tmp/jar/MANIFEST.MF").text.contains("second")
+        executedAndNotSkipped(":jar")
+    }
+
+    def "detects when local state is removed"() {
+        buildFile << """
+            plugins {
+                id 'base'
+            }
+
+            @CacheableTask
+            abstract class WithLocalStateDirectory extends DefaultTask {
+                @LocalState
+                abstract DirectoryProperty getLocalStateDirectory()
+                @OutputFile
+                abstract RegularFileProperty getOutputFile()
+
+                @TaskAction
+                void doStuff() {
+                    def localStateDir = localStateDirectory.get().asFile
+                    localStateDir.mkdirs()
+                    new File(localStateDir, "localState.txt").text = "localState"
+                    outputFile.get().asFile.text = "output"
+                }
+            }
+
+            abstract class WithOutputFile extends DefaultTask {
+                @OutputFile
+                abstract RegularFileProperty getOutputFile()
+
+                @TaskAction
+                void doStuff() {
+                    outputFile.get().asFile.text = "outputFile"
+                }
+            }
+
+            task localStateTask(type: WithLocalStateDirectory) {
+                localStateDirectory = file("build/overlap")
+                outputFile = file("build/localStateOutput.txt")
+            }
+
+            task outputFileTask(type: WithOutputFile) {
+                outputFile = file("build/overlap/outputFile.txt")
+            }
+        """
+        def localStateOutputFile = file("build/localStateOutput.txt")
+
+        when:
+        withWatchFs().withBuildCache().run("localStateTask", "outputFileTask")
+        then:
+        executedAndNotSkipped(":localStateTask", ":outputFileTask")
+        localStateOutputFile.exists()
+
+        when:
+        localStateOutputFile.delete()
+        waitForChangesToBePickedUp()
+        withWatchFs().withBuildCache().run("localStateTask", "outputFileTask")
+        then:
+        skipped(":localStateTask")
+        executedAndNotSkipped(":outputFileTask")
+    }
+
+    // This makes sure the next Gradle run starts with a clean BuildOutputCleanupRegistry
```

Comment:
Why is this required?

Reply:
It is used by `detects when stale outputs are removed` to trigger stale output cleanup.